common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -277,29 +277,30 @@ public void onPermissionRequest(final PermissionRequest request) {
                 // We can only grant or deny permissions for video capture/camera.
                 // To avoid unintentionally granting requests for not defined permissions
                 // we check if the request is for camera.
-                if (!isPermissionRequestForCamera(request)) {
+                if (!CameraPermissionRequest.isValidRequest(request)) {
                     Logger.warn(methodTag, "Permission request is not for camera.");
                     request.deny();
                     return;
                 }
                 // There is a issue in ESTS UX where it sends multiple camera permission requests.
                 // So, if there is already a camera permission request in progress we handle it here.
                 if (mCameraPermissionRequest != null) {
-                    handleRepeatedRequests(request);
+                    Logger.info(methodTag, "Repeated request, granted? " + mCameraPermissionRequest.isGranted());
+                    handleRepeatedCameraRequests(request);
                     return;
                 }
                 Logger.info(methodTag, "New camera request.");
-                mCameraPermissionRequest = request;
-                if (isCameraPermissionGranted()) {
+                mCameraPermissionRequest = new CameraPermissionRequest(request);
+                // The current camera behavior is based on the principle of only prompt once and prompt always.
+                // If the OS level permission is not granted, we show the system prompt.
+                // If the OS level permission was granted previously, we show the rationale to confirm the consent witth the current user.
+                // If the OS level permission was denied previously, we just denny the request.
```

Comment:
nit: denny -> deny

Reply:
also, is there any way to programmatically reset this? (let's say if the user accidentally denies. how do we help them fix that? do we show something to guide them?)

Reply:
We could show a dialog box and send the user to the app info so he can unblock this, but the Android docs disencourage this behavior

https://developer.android.com/training/permissions/requesting
// Explain to the user that the feature is unavailable because
                // the feature requires a permission that the user has denied.
                // At the same time, respect the user's decision. Don't link to
                // system settings in an effort to convince the user to change
                // their decision.